src/libtego/source/user.cpp

```diff
@@ -0,0 +1,38 @@
+#include "error.hpp"
+#include "user.hpp"
+
+tego_user_id::tego_user_id(const tego_v3_onion_service_id_t& serviceId)
+: serviceId(serviceId)
+{ }
+
+void tego_user_id_from_v3_onion_service_id(
+    tego_user_id_t** out_userId,
+    const tego_v3_onion_service_id_t* serviceId,
+    tego_error_t** error)
+{
+    return tego::translateExceptions([=]() -> void
+    {
+        TEGO_THROW_IF_NULL(out_userId);
+        TEGO_THROW_IF_FALSE(*out_userId == nullptr);
```

Comment:
why not `TEGO_THROW_IF_NULL(*out_userId)`? isn't that more readable?

Reply:
We want *out_userId to be NULL to ensure we're not overwriting an existing one. It could be replaced with TEGO_THROW_IF_NOT_NULL(*out_userId) but that macro didn't exist ye when I wrote this code :)

Reply:
ah my bad, i misread that